google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryResultSetImpl.java

```diff
@@ -0,0 +1,554 @@
+          return jsnAry.toString();
+        } else if (currentVal instanceof LocalDateTime) {
+          LocalDateTime dateTime = (LocalDateTime) currentVal;
+          return dateTime.toString();
+        } else {
+          Text textVal = (Text) currentVal;
+          return textVal.toString();
+        }
+      }
+    }
+
+    @Override
+    public String getString(int columnIndex) throws SQLException {
+      if (cursor == null) {
+        return null;
+      } else if (cursor instanceof FieldValueList) {
+        FieldValue fieldValue = ((FieldValueList) cursor).get(columnIndex);
+        return fieldValue.getValue() == null ? null : fieldValue.getStringValue();
+      } else { // Data received from Read API (Arrow)
+        return getString(schemaFieldList.get(columnIndex).getName());
+      }
+    }
+
+    @Override
+    public int getInt(String fieldName) throws SQLException {
+      if (fieldName == null) {
+        throw new SQLException("fieldName can't be null");
+      }
+      if (cursor == null) {
+        return 0; // the column value; if the value is SQL NULL, the value returned is 0 as per
+        // java.sql.ResultSet definition
+      } else if (cursor instanceof FieldValueList) {
+        FieldValue fieldValue = ((FieldValueList) cursor).get(fieldName);
+        return fieldValue.getValue() == null ? 0 : fieldValue.getNumericValue().intValue();
+      } else { // Data received from Read API (Arrow)
+
+        Tuple<Map<String, Object>, Boolean> curTuple = (Tuple<Map<String, Object>, Boolean>) cursor;
+        if (!curTuple.x().containsKey(fieldName)) {
+          throw new SQLException(String.format("Field %s not found", fieldName));
+        }
+        Object curVal = curTuple.x().get(fieldName);
+        if (curVal == null) {
+          return 0;
+        }
+        if (curVal instanceof Text) { // parse from text to int
+          return Integer.parseInt(((Text) curVal).toString());
+        }
+        return ((BigDecimal) curVal).intValue();
+      }
+    }
+
+    @Override
+    public int getInt(int columnIndex) throws SQLException {
+      if (cursor == null) {
+        return 0; // the column value; if the value is SQL NULL, the value returned is 0 as per
+        // java.sql.ResultSet definition
+      } else if (cursor instanceof FieldValueList) {
+        FieldValue fieldValue = ((FieldValueList) cursor).get(columnIndex);
+        return fieldValue.getValue() == null ? 0 : fieldValue.getNumericValue().intValue();
+      } else { // Data received from Read API (Arrow)
+        return getInt(schemaFieldList.get(columnIndex).getName());
+      }
+    }
+
+    @Override
+    public long getLong(String fieldName) throws SQLException {
+      if (fieldName == null) {
+        throw new SQLException("fieldName can't be null");
+      }
+      if (cursor == null) {
+        return 0L; // the column value; if the value is SQL NULL, the value returned is 0 as per
```

Comment:
Not sure the comment is wrong or my understanding of when `cursor` is null is wrong. From the `next()` method, it looked like cursor will be null if result set is exhausted, in which case the column value is not 0.

Reply:
Modified the code to throw a `BigQuerySQLException` in such cases